test-suite/js/src/test/scala/org/scalajs/testsuite/jsinterop/NestedJSClassTest.scala

```diff
@@ -508,6 +510,38 @@ class NestedJSClassTest {
     assertNull(list.tail.tail.tail)
   }
 
+  @Test
+  def jsClassInsideAnonymousClass(): Unit = {
+    val supplier = new Supplier[js.Dynamic] {
+      class InsideAnonymousClass extends js.Object {
+        def x: Int = 1
+      }
+
+      def get(): js.Dynamic =
+        new InsideAnonymousClass().asInstanceOf[js.Dynamic]
+    }
+
+    val obj = supplier.get()
+
+    assertEquals(1, obj.x)
+  }
+
+  @Test
+  def localJSClassInsideAnonymousClass(): Unit = {
+    val supplier = new Supplier[js.Dynamic] {
+      def get(): js.Dynamic = {
+        class LocalInsideAnonymousClass extends js.Object {
+          def x: Int = 1
+        }
+
+        new LocalInsideAnonymousClass().asInstanceOf[js.Dynamic]
+      }
+    }
+
+    val obj = supplier.get()
+
+    assertEquals(1, obj.x)
+  }
```

Comment:
Do we need additional semantic checks on these? (identity of the class?)

Reply:
Not necessarily the identity of the class, but we should definitely check that `obj` has a class other than `js.Object`, and/or that it doesn't have an *own* property named `"x"`.